demos/CoreClrConsoleApplications/HelloWorld/HelloWorld.cs

```diff
@@ -301,4 +314,36 @@ @@@@@@@@@@@         @@@@@@@@@@@@@
         Console.ResetColor();
         Console.WriteLine();
     }
+
+    //HACK: Needed until CoreFX implements Environment.OSVersion
```

Comment:
I realize this is a temporary hack, but it still might be better to have something slightly less hacky, e.g. P/Invoke to uname... if the P/Invoke fails to find the entrypoint, assume you're on Windows, and if it succeeds, use the system name returned to determine what OS you're on.  We've been doing something similar in the corefx tests with https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Interop.PlatformDetection.cs#L45 (this will also switch to the new System.Runtime.InteropServices.RuntimeInformation once it's available via nuget).